### PR TITLE
PRIME-2344 new HA site reg ux

### PR DESCRIPTION
**Pull request by bergomi02**
Please review [PRIME-2346](https://github.com/bcgov/moh-prime/pull/2227) before this PR.

### CODE REVIEW

**Comment by neophyte57** [comment]
I am just beginning the setup as Authorized User and the "progress bar" is showing Completed? 

image

**Comment by bergomi02** [comment]
> I am just beginning the setup as Authorized User and the "progress bar" is showing Completed?
> 
> image

It is fixed.

**Comment by neophyte57** [comment]
I finish entering the PharmaNet Administrator and Technical Support and the screen looks like this (correct)

image

Then I do the following:
1. Click "Site Management" on left
2. Click "Continue Site Submission" for the site I did not submit yet 

Now it shows Completed, which seems to be incorrect 

image

**Comment by neophyte57** [comment]
I tried this on mobile browser (iPhone Safari) and the progress bar isn't displayed.  I think we are supposed to support mobile usage.  It looks like the behavior can also be reproduced using the Developer Tools.

**Comment by bergomi02** [comment]
> I tried this on mobile browser (iPhone Safari) and the progress bar isn't displayed. I think we are supposed to support mobile usage. It looks like the behavior can also be reproduced using the Developer Tools.

I talked to Bryon and this graphic is not supported for mobile use. Especially for site registration, it should be done in desktop. However, the function should work in mobile browser, just no pretty graphic. The graphic will be hidden when the screen size reduced to a certain size.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg.routes.ts [hunk 1]
```diff
@@ -21,6 +21,13 @@ export class HealthAuthSiteRegRoutes {
   public static ADMINISTRATOR = 'administrator';
   public static SITE_OVERVIEW = 'site-overview';
 
+  public static STEP_CARE_TYPE = 'Care Type';
+  public static STEP_SITE_INFORMATION = 'Site Details';
+  public static STEP_HOUR_OPERATION = 'Hours';
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg.routes.ts:26]: Can this be `STEP_HOURS_OPERATION` like `HOURS_OPERATION` a few lines earlier

File: prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import { CompileTemplateMetadata } from '@angular/compiler';
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts:1]: Looks unused

File: prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts [hunk 3]
```diff
@@ -120,11 +133,26 @@ export class ProgressIndicatorComponent implements OnInit, OnChanges, IProgressI
...
     // the input bindings
-    this.updatePercentComplete();
+    if (this.mode === 'percent') {
+      this.updatePercentComplete();
+    }
+    if (this.mode === 'step') {
+      let stepComplete = true;
+      this.steps = this.steps.map(s => {
+        // set isCurrent to true only if in progress.
+        s.isCurrent = s.route === this.currentRoute; //&& this.inProgress;
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts:148]: Remove comments?

**Comment by neophyte57** [comment]
Before I submit 
image

If I edit any of the data using the pencil icons, I get 
image
(incorrect)

File: prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts (hunk 2 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts]: it is fixed.

File: prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg.routes.ts (hunk 1 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/health-auth-site-reg/health-auth-site-reg.routes.ts]: it is fixed.

File: prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts (hunk 3 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/progress-indicator/progress-indicator.component.ts]: it is fixed.

**Comment by neophyte57** [comment]
> > I tried this on mobile browser (iPhone Safari) and the progress bar isn't displayed. I think we are supposed to support mobile usage. It looks like the behavior can also be reproduced using the Developer Tools.
> 
> I talked to Bryon and this graphic is not supported for mobile use. Especially for site registration, it should be done in desktop. However, the function should work in mobile browser, just no pretty graphic. The graphic will be hidden when the screen size reduced to a certain size.

Yes, it does make sense HA site registration should be done on Desktop but let's double-check at Standup.

**Comment by bergomi02** [comment]
> Before I submit image
> 
> If I edit any of the data using the pencil icons, I get image (incorrect)

This should be fixed. However, since we don't keep track of what steps have been completed, the graphic will display as the beginning of the

**Comment by neophyte57** [comment]
I came across the PK_Address issue again.  If you have time, please check it out 
image

**Comment by bergomi02** [comment]
> I came across the PK_Address issue again. If you have time, please check it out image

Very interesting. I logged in and tried to save the address without changes, then I got internal server error. Then I recorded the network and tried again, of course, it worked that time. Please let me know if it happens again.

**Review by neophyte57 — APPROVED** [review]
Neat implementation!

**Pull request merged**